lib/src/client_offline_mixin.dart

```diff
@@ -0,0 +1,310 @@
+  final String model;
+  final String key;
+  final Timestamp accessedAt;
+
+  AccessTimestamp(
+      {required this.model, required this.key, required this.accessedAt});
+
+  factory AccessTimestamp.fromMap(Map<String, Object?> json) => AccessTimestamp(
+        model: json["model"] as String,
+        key: json["key"] as String,
+        accessedAt: json["accessedAt"] as Timestamp,
+      );
+
+  Map<String, Object?> toMap() => {
+        "model": model,
+        "key": key,
+        "accessedAt": accessedAt,
+      };
+}
+
+class ClientOfflineMixin {
+  bool isOnline = true;
+  late Database db;
+
+  StoreRef<String, Map<String, Object?>> _queuedWritesStore =
+      stringMapStoreFactory.store('queuedWrites');
+  StoreRef<String, Map<String, Object?>> _accessTimestampsStore =
+      stringMapStoreFactory.store('accessTimestamps');
+  StoreRef<String, int> _cacheSizeStore = StoreRef<String, int>('cacheSize');
+
+  Random _random = Random();
+
+  Future<void> initOfflineDatabase() async {
+    db = await OfflineDatabase.instance.db();
+  }
+
+  bool resultIsOnline(ConnectivityResult result) {
+    return result == ConnectivityResult.mobile ||
+        result == ConnectivityResult.ethernet ||
+        result == ConnectivityResult.wifi;
+  }
+
+  Future<void> listenForConnectivity() async {
+    void handleConnectivityResult(ConnectivityResult result) {
+      isOnline = resultIsOnline(result);
+    }
+
+    final result = await Connectivity().checkConnectivity();
+    handleConnectivityResult(result);
+    Connectivity().onConnectivityChanged.listen(handleConnectivityResult);
+  }
+
+  String getModel(Uri uri) {
+    final pathSegments = uri.pathSegments;
+    if (pathSegments.length > 5 &&
+        pathSegments[1] == 'databases' &&
+        pathSegments[3] == 'collections' &&
+        pathSegments[5] == 'documents') {
+      return '/' + uri.pathSegments.sublist(0, 6).join('/');
+    }
+
+    if (pathSegments.length > 2 &&
+        pathSegments[1] == 'account' &&
+        pathSegments[2] == 'sessions') {
+      return '/' + uri.pathSegments.sublist(0, 3).join('/');
+    }
+
+    return uri.path;
+  }
+
+  String getKey(Uri uri) {
+    final pathSegments = uri.pathSegments;
+    String key = '';
+    if (pathSegments.length == 2 && pathSegments[1] == 'account') {
+      key = 'current';
+    } else if (pathSegments.length > 6 &&
+        pathSegments[1] == 'databases' &&
+        pathSegments[3] == 'collections' &&
+        pathSegments[5] == 'documents') {
+      key = pathSegments[6];
+    } else if (pathSegments.length > 3 &&
+        pathSegments[1] == 'account' &&
+        pathSegments[2] == 'sessions') {
+      key = pathSegments[3];
+    } else if (pathSegments.length > 2 &&
+        pathSegments[1] == 'account' &&
+        pathSegments[2] == 'prefs') {
+      key = 'current';
+    } else if (pathSegments.length == 2 && pathSegments[1] == 'locale') {
+      key = 'current';
+    }
+
+    return key;
+  }
+
+  // String getId(Uri uri) {
```

Comment:
I would suggest removing large commented out things like this for release.

Reply:
Removed!